### DISCUSSION TITLE
Lesson 12 - Upgrades

### DISCUSSION THREAD

**Discussion by jdonahue94** [smartcontractkit/full-blockchain-solidity-course-py#692] [Q&A]
Hi Team - following along with the tutorial (excellent btw), I am getting the following error when deploying an upgrade to my custom (Agora) contract.

**ProxyAdmin.upgradeAndCall confirmed (Address: low-level delegate call failed)**

Specifically, it seems to occur when adding an initializer to the upgrade function upgrade as such.

`upgrade(account, proxy, agoraV2.address, proxy_admin_contract=proxy_admin, initializer=agoraV2.initialize)`

Note that I was able to successfully encode and initialize my original contract AND when I remove the initializer from the upgrade function everything is 👌🏼. 

Does this make sense: do I not have to call the same initializer on the V2 contract after already have initialized V1? 

Happy to add additional code if needed. 

Thank you,
Jacob

**Comment by cromewar** [comment]
Hello @jdonahue94 actually a tricky question, could you give as a hand here @PatrickAlphaC ?

**Reply by jdonahue94** [reply]
I made a few clarifying edits above.

**Reply by cromewar** [reply]
Ok that's actually better, so the answer is it makes 100% sense when you are upgrading the contract you are actually adding new functionalities to it, a new function for example. but the structure of the contract is pretty much the same so it could be conflicting to have two initializers as you are "upgrading" not "replacing" (not quite literally).

**Reply by jdonahue94** [reply]
Thank you. Well said!
